nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1194,11 +1196,6 @@ def build_train_valid_test_datasets(self):
             test_iters * global_batch_size,
         ]
 
-        if self.trainer.limit_val_batches <= 1.0 and isinstance(self.trainer.limit_val_batches, float):
-            train_valid_test_num_samples[
-                1
-            ] = 1  # This is to make sure we only have one epoch on every validation iteration
-
```

Comment:
Having this is currently causing an index error with blended dataset from mcore. Also I couldn't understand the purpose of having this. Waiting for @shanmugamr1992 's comments on it.

Reply:
Leaving here a comment from an offline conversation with the explanation:

> This is a hack that uses quirk in the dataset implementation that causes the behaviour explained in the comment.
> This array is used to supply the `N` value in the `E = argmin_e e * len(data) >= N` formula. Then `E` is used to construct indices which are then used as an internal representation of dataset. 
>The dataset is not trimmed to `N` after the construction, but to the `E*len(data)` . So setting this to 1, we are sure that `E` is exactly 1, which gives us a single iteration over a validation split.
[...]
In plain words: the number of epochs `E`  is the smallest integer number that when multiplied by length of the data is greater than requested number of samples